### PR TITLE
Scoped repo refactor

### PR DESCRIPTION
**Pull request by RubenGeo**
[AB#32022](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/32022) 

This refactor is needed because we sort of initialized scoped repository twice in the old appoach:

```
 export class RegistrationDataScopedRepository extends ScopedRepository<RegistrationAttributeDataEntity> {
  constructor(
    @Inject(REQUEST) request: ScopedUserRequest,
    @InjectRepository(RegistrationAttributeDataEntity)
    scopedRepository: ScopedRepository<RegistrationAttributeDataEntity>,
  ) {
    super(request, scopedRepository);
  }
```

Here `RegistrationDataScopedRepository` already extents `ScopedRepository` passing it in as argument in the constructor is giving ScopedRepository another ScopedRepository. It's like wrapping it twice.  I think it never gave us issues because the internal defined methods (like 'findOne') are also overwritten twice. 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/src/events/event.repository.ts [hunk 1]
```diff
@@ -15,9 +15,9 @@ export class EventScopedRepository extends ScopedRepository<EventEntity> {
   constructor(
     @Inject(REQUEST) request: ScopedUserRequest,
     @InjectRepository(EventEntity)
-    scopedRepository: ScopedRepository<EventEntity>,
+    repository: ScopedRepository<EventEntity>,
```
Comment (PeterSmallenbroek) [inline services/121-service/src/events/event.repository.ts:18]:
```suggestion
    repository: Repository<EventEntity>,
```

**Review by PeterSmallenbroek — CHANGES_REQUESTED** [review]

File: services/121-service/src/scoped.repository.ts [hunk 2]
```diff
@@ -72,6 +72,11 @@ export class ScopedRepository<T extends ObjectLiteral> extends Repository<T> {
     @Inject(REQUEST) private request: ScopedUserRequest,
     private repository: Repository<T>,
   ) {
+    if (repository instanceof ScopedRepository) {
```
[the remark this answers is not in the document]
Comment (RubenGeo) [reply services/121-service/src/scoped.repository.ts:75]: @aberonni

[the remark this answers is not in the document]
Comment (aberonni) [reply services/121-service/src/scoped.repository.ts:75]: Maybe

```ts

type NotScopedRepository<T extends ObjectLiteral, R = Repository<T>> =
  R extends ScopedRepository<T> ? never : R;

// ... 

private repository: NotScopedRepository<T>,

```

[the remark this answers is not in the document]
Comment (aberonni) [reply services/121-service/src/scoped.repository.ts:75]:
```ts
type NotScopedRepository<R extends Repository<T>, T extends ObjectLiteral> =
  R extends ScopedRepository<T> ? never : R;
  
  // ...
  
  private repository: NotScopedRepository<Repository<T>, T>,
  ```
  
@RubenGeo I can't tell if this works or not - if not, then I'd say it's fine to leave the runtime error

[the remark this answers is not in the document]
Comment (RubenGeo) [reply services/121-service/src/scoped.repository.ts:75]: Unfortunately typescipt is not complaining when I now construct with a ScoperRepository. So merging it now

**Review by PeterSmallenbroek — APPROVED** [review]

**Pull request merged**